### PR TITLE
PAYARA-1402-Upgrade-HK2-to-2.5.0

### PR DESCRIPTION
**Pull request by mulderbaba**
DO NOT MERGE until release of HK2 is finalized

### CODE REVIEW

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

File: appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/deployer/AdministeredObjectDefinitionDeployer.java [hunk 1]
```diff
@@ -336,6 +337,26 @@ public void setEnabled(String value) throws PropertyVetoException {
             return administeredObjectProperties;
         }
 
+        @Override
+        public Property addProperty(Property property) {
+            throw new UnsupportedOperationException();
```
Comment (lprimak) [inline appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/deployer/AdministeredObjectDefinitionDeployer.java]: Should these be actually implemented?

**Comment by mulderbaba** [comment]
jenkins test please

File: appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/deployer/AdministeredObjectDefinitionDeployer.java (hunk 1 above)
Comment (mulderbaba) [inline appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/deployer/AdministeredObjectDefinitionDeployer.java]: @lprimak could be, needs more investigation on the API. I followed the same pattern from resource enable/disable/redeploy methods.

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by arjantijms** [comment]
Jenkins, would you kindly?

**Comment by arjantijms** [comment]
Jenkins, would you kindly?

**Comment by Cousjava** [comment]
jenkins test please

**Comment by arjantijms** [comment]
Jenkins, would you kindly?

**Comment by mulderbaba** [comment]
@arjantijms I think it'd be better to track changes, if we can do rebasing instead of merging

**Comment by arjantijms** [comment]
Jenkins, would you kindly?

**Pull request merged**